cpp/src/arrow/util/io_util.h

```diff
@@ -399,5 +338,12 @@ Status SendSignalToThread(int signum, uint64_t thread_id);
 ARROW_EXPORT
 int64_t GetRandomSeed();
 
+/// \brief Get the current thread id
```

Comment:
It seems odd to put this in io_util.h and not thread_pool.h

Reply:
It's not thread pool-related, it could be useful in other contexts.